src/main/java/io/openshift/booster/catalog/BoosterCatalogService.java

```diff
@@ -69,52 +70,58 @@
 
    private static final Logger logger = Logger.getLogger(BoosterCatalogService.class.getName());
 
-   private final ReentrantReadWriteLock reentrantLock = new ReentrantReadWriteLock();
-
-   private volatile List<Booster> boosters = Collections.emptyList();
+   private volatile Set<Booster> boosters = new ConcurrentSkipListSet<>(Comparator.comparing(Booster::getName));
 
    private final BoosterCatalogPathProvider provider;
+   
+   private final ExecutorService executor;
+   
+   private final CompletableFuture<Set<Booster>> result = new CompletableFuture<>();
+   private volatile boolean indexingStarted = false;
```

Comment:
If using `CompletableFuture.supplyAsync`, you don´t need this variable anymore. Just check if result is null (and remove the `final` and initialization)